core/struct.rbs

```diff
@@ -1,299 +1,51 @@
-  #     # => #<struct PositionalOnly foo={:foo=>1, :bar=>2}, bar=nil>
-  #     # Note that no error is raised, but arguments treated as one hash value
-  #
-  #     # Same as not providing keyword_init:
-  #     Any = Struct.new(:foo, :bar, keyword_init: nil)
-  #     Any.new(foo: 1, bar: 2)
-  #     # => #<struct Any foo=1, bar=2>
-  #     Any.new(1, 2)
-  #     # => #<struct Any foo=1, bar=2>
-  #
-  def initialize: (attribute_name, *attribute_name, ?keyword_init: boolish) ?{ () -> void } -> void
+  def self.members: () -> Array[Symbol]
+  def self.keyword_init?: () -> bool?
 
-  # <!--
-  #   rdoc-file=struct.c
-  #   - each {|value| ... } -> self
-  #   - each -> enumerator
-  # -->
-  # Calls the given block with the value of each member; returns `self`:
-  #
-  #     Customer = Struct.new(:name, :address, :zip)
-  #     joe = Customer.new("Joe Smith", "123 Maple, Anytown NC", 12345)
-  #     joe.each {|value| p value }
-  #
-  # Output:
-  #
-  #     "Joe Smith"
-  #     "123 Maple, Anytown NC"
-  #     12345
-  #
-  # Returns an Enumerator if no block is given.
-  #
-  # Related: #each_pair.
-  #
-  def each: () -> ::Enumerator[Elem?, self]
-          | () { (Elem? item) -> void } -> self
+  def ==: (untyped other) -> bool
+  def eql?: (untyped other) -> bool
+  def hash: () -> Integer
 
-  # <!--
-  #   rdoc-file=struct.c
-  #   - StructClass::members -> array_of_symbols
-  # -->
-  # Returns the member names of the Struct descendant as an array:
-  #
-  #     Customer = Struct.new(:name, :address, :zip)
-  #     Customer.members # => [:name, :address, :zip]
-  #
-  def self.members: () -> ::Array[Symbol]
+  def inspect: () -> String
+  alias to_s inspect
 
-  # <!--
-  #   rdoc-file=struct.c
-  #   - StructClass::keyword_init? -> true or falsy value
-  # -->
-  # Returns `true` if the class was initialized with `keyword_init: true`.
-  # Otherwise returns `nil` or `false`.
-  #
-  # Examples:
-  #     Foo = Struct.new(:a)
-  #     Foo.keyword_init? # => nil
-  #     Bar = Struct.new(:a, keyword_init: true)
-  #     Bar.keyword_init? # => true
-  #     Baz = Struct.new(:a, keyword_init: false)
-  #     Baz.keyword_init? # => false
-  #
-  def self.keyword_init?: () -> (true | false | nil)
+  def to_a: () -> Array[Elem?]
+  def to_h: () -> Hash[Symbol, Elem?]
+          | [K, V] () { (Symbol key, Elem? value) -> [K, V] } -> Hash[K, V]
+  alias values to_a
+
+  def size: () -> Integer
+  alias length size
+
+  def each: () -> Enumerator[Elem?, self]
+          | () { (Elem? value) -> void } -> self
+
+  def each_pair: () -> Enumerator[[Symbol, Elem?], self]
+               | () { ([Symbol, Elem?] key_value) -> void } -> self
+
+  def []: (index name_or_position) -> untyped
+  def []=: (index name_or_position, Elem value) -> Elem
+         | (index name_or_position, nil) -> nil
+
+  def select: () -> Enumerator[Elem?, Array[Elem?]]
+            | () { (Elem? value) -> boolish } -> Array[Elem?]
+
+  alias filter select
```

Comment:
do these have the same signatures as the `select` from `Enumerable[Elem?]`?

Reply:
No. Since  you can do `My3dPoint.new(1)`, the extra elements are assigned as `nil`. Thus, anything that interacts with them are `Elem?`.

Reply:
Yes, but with `Enumerable[Elem?]`, everything from `Enumerable` are already `Elem?`s. Here I’m pointing out that by applying `Elem?` to `Enumerable`’s type variable `Elem`, both types have the same `#select`, `#filter` and `#to_a`.

Reply:
Ah, yes. However, i'm, annotating any methods that are explicitly defined in the base class and not inherited

Reply:
Detailing what’s inherited and what’s overriden isn’t RBS’s job…